server/src/main/java/org/opensearch/search/resource_limit_group/tracker/ResourceLimitsGroupResourceUsageTrackerService.java

```diff
@@ -0,0 +1,300 @@
+                .toMap();
+
+            resourceUsage.put(resourceLimitGroupName, resourceLimitGroupUsage);
+
+        }
+    }
+
+    // @Override
+    // public SandboxStatsHolder getStats() {
+    // return null;
+    // }
+
+    private AbsoluteResourceUsage calculateAbsoluteResourceUsageFor(Task task) {
+        TaskResourceUsage taskResourceUsage = task.getTotalResourceStats();
+        long cpuTimeInNanos = taskResourceUsage.getCpuTimeInNanos();
+        long jvmAllocations = taskResourceUsage.getMemoryInBytes();
+        long taskElapsedTime = timeNanosSupplier.getAsLong() - task.getStartTimeNanos();
+        return new AbsoluteResourceUsage(
+            (cpuTimeInNanos * 1.0f) / (taskElapsedTime * numberOfAvailableProcessors),
+            ((jvmAllocations * 1.0f) / totalAvailableJvmMemory)
+        );
+    }
+
+    /**
+     * Value holder class for resource usage in absolute terms with respect to system/process mem
+     */
+    private static class AbsoluteResourceUsage {
+        private final double absoluteCpuUsage;
+        private final double absoluteJvmAllocationsUsage;
+
+        public AbsoluteResourceUsage(double absoluteCpuUsage, double absoluteJvmAllocationsUsage) {
+            this.absoluteCpuUsage = absoluteCpuUsage;
+            this.absoluteJvmAllocationsUsage = absoluteJvmAllocationsUsage;
+        }
+
+        public static AbsoluteResourceUsage merge(AbsoluteResourceUsage a, AbsoluteResourceUsage b) {
+            return new AbsoluteResourceUsage(
+                a.absoluteCpuUsage + b.absoluteCpuUsage,
+                a.absoluteJvmAllocationsUsage + b.absoluteJvmAllocationsUsage
+            );
+        }
+
+        public double getAbsoluteCpuUsageInPercentage() {
+            return absoluteCpuUsage * 100;
+        }
+
+        public double getAbsoluteJvmAllocationsUsageInPercent() {
+            return absoluteJvmAllocationsUsage * 100;
+        }
+
+        /**
+         *
+         * @return {@code Map<String, Double>}
+         * which captures key as resource and value as the percentage value
+         */
+        public Map<String, Double> toMap() {
+            Map<String, Double> map = new HashMap<>();
+            // We can put the additional resources into this map in the future
+            map.put(JVM, getAbsoluteJvmAllocationsUsageInPercent());
+            return map;
+        }
+    }
+
+    /**
+     * filter out the deleted sandboxes which still has unfi
+     */
+    public void pruneResourceLimitGroup() {
+        toDeleteResourceLimitGroups = toDeleteResourceLimitGroups.stream().filter(this::hasUnfinishedTasks).collect(Collectors.toList());
+    }
+
+    private boolean hasUnfinishedTasks(String sandboxId) {
+        return false;
+    }
+
+    /**
+     * method to handle the completed tasks
+     * @param task represents completed task on the node
+     */
+    @Override
+    public void onTaskCompleted(Task task) {}
+
```

Comment:
If task completion is overridden, we don't need to stream the list of deleted RLGs? Although, should not matter given RLG deletion should be rare